src/addrman.cpp

```diff
@@ -1016,6 +1021,35 @@ std::optional<AddressPosition> AddrManImpl::FindAddressEntry_(const CAddress& ad
     }
 }
 
+bool AddrManImpl::HasFewReachableAddresses_(const std::unordered_set<Network>& networks)
+{
+    AssertLockHeld(cs);
+
+    if (m_network_counts.empty()) {
+        return false;
+    }
+
+    int total_size{0};
+    int reachable_size{0};
+
+    for (const auto& item : m_network_counts) {
+        const auto& net{item.first};
+        const auto& count{item.second};
+
+        total_size += (count.n_new + count.n_tried);
+
+        if (networks.count(net)) {
```

Comment:
nit: [`contains`](https://en.cppreference.com/w/cpp/container/unordered_set/contains) is equivalent but more semantically correct here